### PR TITLE
Allow callout users to be revoked

### LINKED ISSUE #5477: Decentralized auth callout: Connections of revoked users are not closed immediately
### Observed behavior

We are using a decentralized auth callout handler [similar to the nats-by-example service](https://github.com/ConnectEverything/nats-by-example/blob/main/examples/auth/callout-decentralized/cli/service/main.go).

Following relevant accounts & users are in the system:

- `service-account` (auth-callout encryption target)
    - `sentinel-user`
- `customer-account`
    - `customer-user`

Customer accounts like `customer-account` are created in an automated manner. The signing keys for these accounts are stored in a KV for the auth callout handling service.

NATS clients are connecting to the NATS server using the `sentinel-user` and token info including the account ID like `customer-account`. The NATS server is requesting an auth callout to the handling service. The auth callout handling service is creating a `customer-user` and allowing the initial connection request.

Now `customer-user` is being added to the revocations list of `customer-account`. The `customer-account` JWT having `customer-user` in it's revocation list is then pushed to the NATS server. The initial connection from the NATS client remains open.

### Expected behavior

As with statically created users, it is expected that once the updated account JWT is pushed to the NATS server, the open connection from a user in the accounts revocation list is immediately closed ([as documented here for example](https://docs.nats.io/using-nats/nats-tools/nsc/revocation#pushing-the-changes-to-the-nats-servers)).

### Server and client version

Server v2.10.16
CLI Client v0.1.4 and nats.go v1.35.0

### Host environment

_No response_

### Steps to reproduce

_No response_

### PR DESCRIPTION
**Pull request by aricart**
[FIX] added logic to enable users generated by a callout to be revoked - the callout is responsible to map the user ids for revocation purposes.

Fix #5477

### CODE REVIEW

File: server/accounts.go [hunk 1]
```diff
@@ -3686,6 +3686,14 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 		}
 		c.mu.Lock()
 		c.applyAccountLimits()
+		// if we have an nkey user we are a callout user - save
+		// the issuedAt, and nkey user id to honor revocations
+		nkeyUserID := ""
```
Comment (derekcollison) [inline server/accounts.go:3691]: var nkeyUserID string

File: server/accounts.go [hunk 2]
```diff
@@ -3686,6 +3686,14 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
...
+		issuedAt := int64(0)
```
Comment (derekcollison) [inline server/accounts.go:3692]: vat issuedAt uint64

File: server/accounts.go [hunk 3]
```diff
@@ -3704,6 +3712,27 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 				continue
 			}
 		}
+
+		// if we extracted nkeyUserID and issuedAt we are a callout user
+		// calloutIAT should only be set if we are in callout type scenario
```
Comment (derekcollison) [inline server/accounts.go:3717]: in a callout type..

File: server/accounts.go [hunk 4]
```diff
@@ -3704,6 +3712,27 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
...
+		// the user JWT is _NOT_ associated with the client for callouts,
+		// so we rely on the calloutIAT to know when the JWT was issued
+		// revocations simply state that JWT issued before or by that date
+		// are not valid
+		if ac.Revocations != nil && nkeyUserID != "" && issuedAt > 0 {
```
Comment (derekcollison) [inline server/accounts.go:3722]: nkeyUserID != _EMPTY_

File: server/auth.go [hunk 5]
```diff
@@ -59,7 +59,9 @@ type ClientAuthentication interface {
 
 // NkeyUser is for multiple nkey based users
 type NkeyUser struct {
-	Nkey                   string              `json:"user"`
+	Nkey string `json:"user"`
+	// this is a copy of the issued at (iat) field in the jwt
```
Comment (derekcollison) [inline server/auth.go:63]: PLace comment at end to keep spacing same.

**Review by derekcollison — CHANGES_REQUESTED** [review]
LGTM - a few small nits.

File: server/accounts.go (hunk 1 above)
Comment (aricart) [reply server/accounts.go]: Fixed

File: server/accounts.go (hunk 3 above)
Comment (aricart) [reply server/accounts.go]: Fixed

File: server/auth.go (hunk 5 above)
Comment (aricart) [reply server/auth.go]: Fixed

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**